src/main/java/click/escuela/grade/service/impl/GradeServiceImpl.java

```diff
@@ -81,6 +81,27 @@ public GradeDTO getById(String id) throws TransactionException {
 	public List<GradeDTO> getByStudentId(String studentId) {
 		return Mapper.mapperToGradesDTO(gradeRepository.findByStudentId(UUID.fromString(studentId)));
 	}
+	
+	public List<Grade> getByStudentIds(List<String> studetnIds) {
+		List<UUID> listUUID = studetnIds.stream().map(UUID::fromString).collect(Collectors.toList());
+		return gradeRepository.findByStudentIdIn(listUUID);
+	}
+	
+	public List<StudentShortDTO> getStudentsWithGrades(List<StudentShortDTO> students) {
+		List<String> studentIds = students.stream().map(StudentShortDTO::getId).collect(Collectors.toList());
+		List<Grade> grades = getByStudents(studentIds);
+		students.forEach(student -> student.setGrades(Mapper.mapperToGradesDTO(getGradesByStudent(grades, student))));
+		return students;
+	}
+	
+	private List<Grade> getByStudents(List<String> studentsIds) {
+		List<UUID> listUUID = studentsIds.stream().map(UUID::fromString).collect(Collectors.toList());
+		return gradeRepository.findByStudentIdIn(listUUID);
+	}
+	
+	private List<Grade> getGradesByStudent(List<Grade> grades, StudentShortDTO student) {
+		return grades.stream().filter(grade -> grade.getStudentId().equals(UUID.fromString(student.getId()))).collect(Collectors.toList());
```

Comment:
if you extract the code inside the filter the code can be more readable
private List<Grade> getGradesByStudent(List<Grade> grades, StudentShortDTO student) {
	return grades.stream().filter(studentIdOfGradeIsEquals(student)).collect(Collectors.toList());
}
private Predicate<? super Grade> studentIdOfGradeIsEquals(StudentShortDTO student) {
	return grade -> grade.getStudentId().equals(UUID.fromString(student.getId()));
}
the method name is horrible you change it